### ISSUE TITLE
Too many wrong region proposal detection

### ISSUE THREAD

**Issue by ranju2015** [rbgirshick/py-faster-rcnn#497]
![weepingtoadfish_frame_0050](https://cloud.githubusercontent.com/assets/14028800/23148443/86644874-f831-11e6-9dbc-9121df89bc72.jpg)

Hi friend, I have successfully trained my system with 4074 annotated regions and 40 classes. My total iteration was 70k. But when I test get detection and classification results like above. I appreciate your feedback if you have already gone through such problems.

**Comment by ranju2015** [comment]
The problem stated above has been improved by setting __C.TEST.BBOX_REG = False during testing in config.py. However, the detected bounding boxes are not perfect. Please have a look at the image below.
![weepingtoadfish_frame_0660](https://cloud.githubusercontent.com/assets/14028800/23331783/b254016a-fbb8-11e6-84da-2d030d50210c.jpg)

Guys I need your suggestions for some improvement. I am using the ZF model. and done 100k iteration. The total number of class is 40 and and training configuration is given below.

+ echo Logging output to experiments/logs/faster_rcnn_end2end_ZF_.txt.2017-02-25_00-48-30
Logging output to experiments/logs/faster_rcnn_end2end_ZF_.txt.2017-02-25_00-48-30
+ ./tools/train_net.py --gpu 0 --solver models/pascal_voc/ZF/faster_rcnn_end2end/solver.prototxt --weights data/imagenet_models/ZF.v2.caffemodel --imdb voc_2007_trainval --iters 250000 --cfg experiments/cfgs/faster_rcnn_end2end.yml
Called with args:
Namespace(cfg_file='experiments/cfgs/faster_rcnn_end2end.yml', gpu_id=0, imdb_name='voc_2007_trainval', max_iters=250000, pretrained_model='data/imagenet_models/ZF.v2.caffemodel', randomize=False, set_cfgs=None, solver='models/pascal_voc/ZF/faster_rcnn_end2end/solver.prototxt')
Using config:
{'DATA_DIR': '/home/s2830250/py-faster-rcnn-40/data',
 'DEDUP_BOXES': 0.0625,
 'EPS': 1e-14,
 'EXP_DIR': 'faster_rcnn_end2end',
 'GPU_ID': 0,
 'MATLAB': 'matlab',
 'MODELS_DIR': '/home/s2830250/py-faster-rcnn-40/models/pascal_voc',
 'PIXEL_MEANS': array([[[ 102.9801,  115.9465,  122.7717]]]),
 'RNG_SEED': 3,
 'ROOT_DIR': '/home/s2830250/py-faster-rcnn-40',
 'TEST': {'BBOX_REG': False,
          'HAS_RPN': True,
          'MAX_SIZE': 1000,
          'NMS': 0.3,
          'PROPOSAL_METHOD': 'selective_search',
          'RPN_MIN_SIZE': 16,
          'RPN_NMS_THRESH': 0.7,
          'RPN_POST_NMS_TOP_N': 300,
          'RPN_PRE_NMS_TOP_N': 6000,
          'SCALES': [600],
          'SVM': False},
 'TRAIN': {'ASPECT_GROUPING': True,
           'BATCH_SIZE': 128,
           'BBOX_INSIDE_WEIGHTS': [1.0, 1.0, 1.0, 1.0],
           'BBOX_NORMALIZE_MEANS': [0.0, 0.0, 0.0, 0.0],
           'BBOX_NORMALIZE_STDS': [0.1, 0.1, 0.2, 0.2],
           'BBOX_NORMALIZE_TARGETS': True,
           'BBOX_NORMALIZE_TARGETS_PRECOMPUTED': True,
           'BBOX_REG': True,
           'BBOX_THRESH': 0.5,
           'BG_THRESH_HI': 0.5,
           'BG_THRESH_LO': 0.0,
           'FG_FRACTION': 0.25,
           'FG_THRESH': 0.5,
           'HAS_RPN': True,
           'IMS_PER_BATCH': 1,
           'MAX_SIZE': 1000,
           'PROPOSAL_METHOD': 'gt',
           'RPN_BATCHSIZE': 256,
           'RPN_BBOX_INSIDE_WEIGHTS': [1.0, 1.0, 1.0, 1.0],
           'RPN_CLOBBER_POSITIVES': False,
           'RPN_FG_FRACTION': 0.5,
           'RPN_MIN_SIZE': 16,
           'RPN_NEGATIVE_OVERLAP': 0.3,
           'RPN_NMS_THRESH': 0.7,
           'RPN_POSITIVE_OVERLAP': 0.7,
           'RPN_POSITIVE_WEIGHT': -1.0,
           'RPN_POST_NMS_TOP_N': 2000,
           'RPN_PRE_NMS_TOP_N': 12000,
           'SCALES': [600],
           'SNAPSHOT_INFIX': '',
           'SNAPSHOT_ITERS': 10000,
           'USE_FLIPPED': True,
           'USE_PREFETCH': False},
 'USE_GPU_NMS': True}
Loaded dataset `voc_2007_trainval` for training
Set proposal method: gt
Appending horizontally-flipped training examples...
voc_2007_trainval gt roidb loaded from /home/s2830250/py-faster-rcnn-40/data/cache/voc_2007_trainval_gt_roidb.pkl
done
Preparing training data...
done
4074 roidb entries
Output will be saved to `/home/s2830250/py-faster-rcnn-40/output/faster_rcnn_end2end/voc_2007_trainval`
Filtered 0 roidb entries: 4074 -> 4074
Computing bounding-box regression targets...

**Comment by andrewsilva9** [comment]
I was struggling with the same thing and setting that flag helped tremendously! Thanks!

**Comment by sodeypunk** [comment]
Yep, same issue here on the INRIA dataset. Would like to understand why as well.

**Comment by sodeypunk** [comment]
Yes, i had the same issue as well. Not sure why the demo script didnt have this issue.

**Comment by boyflytobeman** [comment]
@sodeypunk @andrewsilva9 ,if __C.TEST.BBOX_REG = False ,the detection accuracy will drop, any other solution?

**Comment by aum12** [comment]
@rbgirshick @ranju2015 Do you have any intuition as to why the spurious proposals are eliminated once bbox regression is disabled? 
It's observably clear that this change will impact localization as predictions are using the default anchor box scale/aspect ratio. But I'm not sure why this setting is removing the false positives?

**Comment by Pezaun** [comment]
Same problem for me. Any body have some intuition to solve this issue? Recently I train an VGG16 in my own dataset and the detection with __C.TEST.BBOX_REG = True was terrible. When __C.TEST.BBOX_REG = False it is acceptable, mas not impressive such as described by the papers.

Regards!
